var/spack/repos/builtin/packages/fenics-dolfin/package.py

```diff
@@ -0,0 +1,148 @@
+    depends_on('cmake@3.5:', type='build')
+
+    depends_on('py-setuptools', type='build', when='+python')
+    depends_on('py-sphinx@1.0.1:', type='build', when='+python+doc')
+
+    def url_for_version(self, version):
+        url = "https://bitbucket.org/fenics-project/dolfin/get"
+        if version >= Version('2017.1.0'):
+            url += "/{0}.tar.gz".format(version)
+        else:
+            url += "/dolfin-{0}.tar.gz".format(version)
+        return url
+
+    def cmake_is_on(self, option):
+        return 'ON' if option in self.spec else 'OFF'
+
+    def cmake_args(self):
+        return [
+            '-DDOLFIN_ENABLE_DOCS:BOOL={0}'.format(
+                self.cmake_is_on('+doc')),
+            '-DBUILD_SHARED_LIBS:BOOL={0}'.format(
+                self.cmake_is_on('+shared')),
+            '-DDOLFIN_SKIP_BUILD_TESTS:BOOL=ON',
+            '-DDOLFIN_ENABLE_OPENMP:BOOL={0}'.format(
+                self.cmake_is_on('+openmp')),
+            '-DDOLFIN_ENABLE_CHOLMOD:BOOL={0}'.format(
+                self.cmake_is_on('suite-sparse')),
+            '-DDOLFIN_ENABLE_HDF5:BOOL={0}'.format(
+                self.cmake_is_on('hdf5')),
+            '-DDOLFIN_ENABLE_MPI:BOOL={0}'.format(
+                self.cmake_is_on('mpi')),
+            '-DDOLFIN_ENABLE_PARMETIS:BOOL={0}'.format(
+                self.cmake_is_on('parmetis')),
+            '-DDOLFIN_ENABLE_PASTIX:BOOL={0}'.format(
+                self.cmake_is_on('pastix')),
+            '-DDOLFIN_ENABLE_PETSC:BOOL={0}'.format(
+                self.cmake_is_on('petsc')),
+            '-DDOLFIN_ENABLE_PETSC4PY:BOOL={0}'.format(
+                self.cmake_is_on('py-petsc4py')),
+            '-DDOLFIN_ENABLE_PYTHON:BOOL={0}'.format(
+                self.cmake_is_on('python')),
+            '-DDOLFIN_ENABLE_QT:BOOL={0}'.format(
+                self.cmake_is_on('qt')),
+            '-DDOLFIN_ENABLE_SCOTCH:BOOL={0}'.format(
+                self.cmake_is_on('scotch')),
+            '-DDOLFIN_ENABLE_SLEPC:BOOL={0}'.format(
+                self.cmake_is_on('slepc')),
+            '-DDOLFIN_ENABLE_SLEPC4PY:BOOL={0}'.format(
+                self.cmake_is_on('py-slepc4py')),
+            '-DDOLFIN_ENABLE_SPHINX:BOOL={0}'.format(
+                self.cmake_is_on('py-sphinx')),
+            '-DDOLFIN_ENABLE_TRILINOS:BOOL={0}'.format(
+                self.cmake_is_on('trilinos')),
+            '-DDOLFIN_ENABLE_UMFPACK:BOOL={0}'.format(
+                self.cmake_is_on('suite-sparse')),
+            '-DDOLFIN_ENABLE_VTK:BOOL={0}'.format(
+                self.cmake_is_on('vtk')),
+            '-DDOLFIN_ENABLE_ZLIB:BOOL={0}'.format(
+                self.cmake_is_on('zlib')),
+        ]
+
+    @run_after('install')
+    def install_python_interface(self):
+        if '+python' in self.spec:
+            if self.version >= Version('2018.1.0'):
+                cd('python')
+                python('setup.py', 'install', '--prefix={0}'.format(self.prefix))
```

Comment:
Flake is complaining about line length here

Reply:
Thanks @scheibelp, I'll split this line when I figure out how to get this to work. Currently, CMake fails to find the DOLFIN package (see above).

Reply:
@benzwick did you ever figure out what's wrong with CMake? I tried your PR and even with 2019.1.0 version the error is the same...

Reply:
```
diff --git a/var/spack/repos/builtin/packages/fenics-dolfin/package.py b/var/spack/repos/builtin/packages/fenics-dolfin/package.py
index 539a8afac..b5e7ef6d1 100644
--- a/var/spack/repos/builtin/packages/fenics-dolfin/package.py
+++ b/var/spack/repos/builtin/packages/fenics-dolfin/package.py
@@ -155,9 +155,12 @@ class FenicsDolfin(CMakePackage):
 
         return opts
 
+    def setup_environment(self, spack_env, run_env):
+        spack_env.set('DOLFIN_DIR', self.prefix)
+
     @run_after('install')
     def install_python_interface(self):
         if '+python' in self.spec:
             if self.version >= Version('2018.1.0'):
                 cd('python')
-                python('setup.py', 'install', '--prefix={0}'.format(self.prefix))
+                python('setup.py', 'install', '--single-version-externally-managed', '--root=/', '--prefix={0}'.format(self.prefix))
```
seems to fix the problem. Adopted from https://github.com/JaroslavHron/spack_repo/blob/master/fenics/packages/dolfin/package.py

Reply:
@davydden no, I didn't get this working. I just build FEniCS maually from source instead. I'm happy if you want to make these changes. Everything should work now as the issue with Python has also been resolved as far as I know.